core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -409,6 +421,15 @@ private void install(Terminal terminal, boolean isBatch, Path tmpRoot) throws Ex
             }
 
             Files.move(tmpRoot, destination, StandardCopyOption.ATOMIC_MOVE);
+            try (DirectoryStream<Path> stream = Files.newDirectoryStream(destination)) {
+                for (Path pluginFile : stream) {
+                    if (Files.isDirectory(pluginFile)) {
+                        Files.setPosixFilePermissions(pluginFile, dirAndExecutablePerms);
```

Comment:
Sadly, I think that these need to be guarded (e.g., Windows does not have POSIX support).